hack/ocp-rebase.sh

```diff
@@ -8,24 +8,52 @@
 
 set -e
 
+# make sure we have downstream and upstream trees before the script starts
+if ! git remote show | grep -q ^upstream$; then
+    git remote add upstream https://github.com/nmstate/kubernetes-nmstate
+fi
+
+if ! git remote show | grep -q ^downstream$; then
+    git remote add downstream https://github.com/openshift/kubernetes-nmstate
+fi
+
 git remote update
 
 upstream_remote=$(git remote -v | grep "https://github.com/nmstate/kubernetes-nmstate (fetch)" | awk '{print $1}')
 downstream_remote=$(git remote -v | grep "https://github.com/openshift/kubernetes-nmstate (fetch)" | awk '{print $1}')
 
-read -p "Source branch from upstream (usually main): " upstream_source_branch
+read -p "Source branch from upstream [default: main]: " upstream_source_branch
+if [ -z "${upstream_source_branch}" ]
+then
+    upstream_source_branch="main"
+fi
+echo "Selected branch from upstream: ${upstream_source_branch}"
+
 if ! git show-ref --quiet ${upstream_remote}/${upstream_source_branch}; then
     echo "Branch ${upstream_source_branch} does not exist in ${upstream_remote}"
     exit 1
 fi
 
-read -p "Target branch in downstream (usually master): " downstream_target_branch
+read -p "Target branch in downstream [default: master]: " downstream_target_branch
+if [ -z "${downstream_target_branch}" ]
+then
+    downstream_target_branch="master"
+fi
+echo "Selected branch from downstream: ${downstream_target_branch}"
+
 if ! git show-ref --quiet ${downstream_remote}/${downstream_target_branch}; then
     echo "Branch ${downstream_target_branch} does not exist in ${downstream_remote}"
     exit 1
 fi
 
-read -p "CommitId of latest merge commit in ${downstream_remote}/${downstream_target_branch}: " last_merge_commit
+commitID=$(git log --oneline downstream/master | grep "merge nmstate/main" | cut -d ' ' -f 1 | head -n 1)
```

Comment:
We shouldn't hardcode the remote and branch names. downstream/master should be ${downstream_remote}/${downstream_target_branch}.

Also, are we sure it's always going to be "merge nmstate/main"? Looks like the commit message is based on upstream_source_branch.

Reply:
merge nmstate/${upstream_source_branch}